spec/services/proofing/aamva/proofer_spec.rb

```diff
@@ -195,6 +207,156 @@
+          match_indicator_name: 'AddressCityMatchIndicator',
+          required_part_of: :address,
+        },
+        state: {
+          required: false,
+          match_indicator_name: 'AddressStateCodeMatchIndicator',
+          required_part_of: :address,
+        },
+        zipcode: {
+          required: false,
+          match_indicator_name: 'AddressZIP5MatchIndicator',
+          required_part_of: :address,
+        },
+        dob: {
+          required: true,
+          match_indicator_name: 'PersonBirthDateMatchIndicator',
+        },
+        state_id_issued: {
+          required: false,
+          match_indicator_name: 'DriverLicenseIssueDateMatchIndicator',
+        },
+        state_id_number: {
+          required: true,
+          match_indicator_name: 'DriverLicenseNumberMatchIndicator',
+        },
+        state_id_expiration: {
+          required: false,
+          match_indicator_name: 'DriverLicenseExpirationDateMatchIndicator',
+        },
+        state_id_type: {
+          required: false,
+          match_indicator_name: 'DocumentCategoryMatchIndicator',
+        },
+        first_name: {
+          required: true,
+          match_indicator_name: 'PersonFirstNameExactMatchIndicator',
+        },
+        last_name: {
+          required: true,
+          match_indicator_name: 'PersonLastNameExactMatchIndicator',
+        },
+      }.each do |(attribute, options)|
+        describe attribute.to_s do
+          context 'when attribute is not verified' do
+            let(:verification_response) do
+              XmlHelper.modify_xml_at_xpath(
+                super(),
+                "//#{options[:match_indicator_name]}",
+                'false',
+              )
+            end
+
+            if options[:required]
+              it 'results in failure' do
+                expect(result.success?).to be false
+              end
+            else
+              it 'still succeeds' do
+                expect(result.success?).to be true
+              end
+            end
+
+            if options[:optional_part_of]
+              it "doesn't stop #{options[:optional_part_of]} appearing in requested_attributes" do
+                expect(result.requested_attributes).to include(options[:optional_part_of] => 1)
+              end
+              it "doesn't stop #{options[:optional_part_of]} appearing in verified_attributes" do
+                expect(result.verified_attributes).to include(options[:optional_part_of])
+              end
+            elsif options[:required_part_of]
+              it "doesn't stop #{options[:required_part_of]} appearing in requested_attributes" do
+                expect(result.requested_attributes).to include(options[:required_part_of] => 1)
+              end
+              it "stops #{options[:required_part_of]} appearing in verified_attributes" do
+                expect(result.verified_attributes).not_to include(options[:required_part_of])
+              end
+            else
```

Comment:
I wonder if we could flip/invert this test style?

One tool is shared examples (which I don't think we could use but we could), in the spirit of leveraging RSpec.

On the other hand, we could probably just do like plain methods?

```ruby
describe '#state_id_number' do
  test_required_attribute(attribute: 'state_id_number', match_indicator: 'DriverLicenseNumberMatchIndicator')
end

describe '#city' do
  test_required_address_attribute(attribute: 'city', match_indicator: 'AddressCityMatchIndicator')
end

describe('#state_id_issued') do
  test_optional_attribute(attribute: 'state_id_issued', match_indicator: 'DriverLicenseIssueDateMatchIndicator')
end

def self.test_required_attribute(attribute:, match_indicator:)
  it '...' do
  end
end
```

Reply:
Yeah, I like that suggestion. I was struggling with how to get the coverage without having to type it all out, not sure why I didn't just think of methods